### PR TITLE
VS handler fix for 0.10.6

### PR DESCRIPTION
**Pull request by bieniekmateusz**
This removes a bug which is the dependence on the order of the list. Ie when param_is_list is set to true, it stays true for each next handler. 

Is one way to addresses #1422 with minimal changes

- [ ] Tag issue being addressed
- [ ] Add [tests](https://github.com/openforcefield/openff-toolkit/tree/main/openff/toolkit/tests)
- [ ] Update docstrings/[documentation](https://github.com/openforcefield/openff-toolkit/tree/main/docs), if applicable
- [ ] [Lint](https://open-forcefield-toolkit.readthedocs.io/en/latest/developing.html#style-guide) codebase
- [ ] Update [changelog](https://github.com/openforcefield/openff-toolkit/blob/main/docs/releasehistory.rst)

### CODE REVIEW

File: openff/toolkit/typing/engines/smirnoff/forcefield.py [hunk 1]
```diff
@@ -1507,6 +1503,8 @@ def label_molecules(self, topology):
                 # Now make parameter_matches into a dict mapping
                 # match objects to ParameterTypes
 
+                param_is_list = issubclass(type(parameter_handler), VirtualSiteHandler)
```
Comment (mattwthompson) [inline openff/toolkit/typing/engines/smirnoff/forcefield.py:1506]: Perhaps
```suggestion
                param_is_list = isinstance(parameter_handler, VirtualSiteHandler)
```

**Comment by mattwthompson** [comment]
Thanks for the fix - could you add a unit test(s) to ensure this fixes the linked issue?

**Comment by mattwthompson** [comment]
Could you also modify these files to have `openff-interchange <0.2`? This is unrelated to your changes; the branch was previously failing

* `devtools/conda-envs/test_env.yaml`
* `devtools/conda-envs/rdkit.yaml`
* `devtools/conda-envs/openeye.yaml`

**Comment by mattwthompson** [comment]
One more to get the environments updated: please add `qcportal >= 0.15` to the same files - thanks and sorry for our upstream being out of date

The `OpenEye=True` builds will all fail because the license is not accessible from a fork, so just ignore those runs

**Review by j-wags — COMMENTED** [review]
This is a small enough change that we can go forward without a test if @bieniekmateusz doesn't have time right now. Then we could cut a 0.10.7 release with this change! I'll let @mattwthompson make the final call here, just adding my two cents.

**Comment by bieniekmateusz** [comment]
Thanks, I have tested it locally as part of our pipeline and in a small script as suggested by @JoshHorton , and with a small snippet:
```
from qubekit.cli.utils import LocalCoordinateVirtualSiteHandler
from openff.toolkit.typing.engines.smirnoff import ForceField
from openff.toolkit.topology import Molecule

methanol = Molecule.from_smiles("CO")
ff = ForceField("methanol.offxml", load_plugins=True)
ff.label_molecules(methanol.to_topology())
```
However, when I extract the class and use it directly for a test case, it returns a different structure/model, so I haven't had a look at that yet.

**Comment by bieniekmateusz** [comment]
Thank you both and @jthorton, I added a naive test case that checks if the `param_is_list` is triggered.

**Comment by mattwthompson** [comment]
Thanks! I double-checked to see that the test fails with 0.10.6.

**Pull request merged**